modules/apps/frontend-js/frontend-js-translation-web/src/main/resources/META-INF/resources/TranslationModal.js

```diff
@@ -0,0 +1,201 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import {ClayButtonWithIcon} from '@clayui/button';
+import ClayDropDown from '@clayui/drop-down';
+import {ClayInput} from '@clayui/form';
+import ClayIcon from '@clayui/icon';
+import ClayLabel from '@clayui/label';
+import ClayManagementToolbar from '@clayui/management-toolbar';
+import ClayModal from '@clayui/modal';
+import ClayTable from '@clayui/table';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+const TranslationModal = ({
+	ariaLabels = {
+		default: Liferay.Language.get('default'),
+		manageTranslations: Liferay.Language.get('manage-translations'),
+		translated: Liferay.Language.get('translated'),
+		untranslated: Liferay.Language.get('untranslated'),
+	},
+	defaultLocale,
+	locales,
+	observer,
+	translations,
+}) => {
+	const [active, setActive] = React.useState(false);
+	const [icon, setIcon] = React.useState('search');
+	const inputRef = React.createRef();
```

Comment:
Why keep the input uncontrolled? we could just keep the component controlled instead of using a reference to change the component value in some cases, it is usually preferable if you use the controlled component strategy, so you are free to make strategies like the previous comment.

https://reactjs.org/docs/uncontrolled-components.html

Reply:
I agree with both what @bryceosterhaus and you said, there's no need for the input to be uncontrolled, my mistake

Reply:
Updated in [`8a9dfad`](https://github.com/liferay-frontend/liferay-portal/pull/1184/commits/8a9dfad58eeeec19dfd11eed9e325170680b3266) - sorry this might be the "hardest" thing to review, but it does look nicer